plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeMetadataSchemas.java

```diff
@@ -0,0 +1,349 @@
+                    .length(PRECISION_TO_BYTE_COUNT[trinoDecimalType.getPrecision()])
+                    .as(decimalType(trinoDecimalType.getScale(), trinoDecimalType.getPrecision()));
+            return buildType(name, id, parent, typeBuilder, trinoType, primitiveTypesConsumer);
+        }
+        switch (primitiveType) {
+            case "string" -> {
+                typeBuilder = Types.primitive(PrimitiveType.PrimitiveTypeName.BINARY, repetition).as(LogicalTypeAnnotation.stringType());
+                trinoType = VARCHAR;
+            }
+            case "byte" -> {
+                typeBuilder = Types.primitive(PrimitiveType.PrimitiveTypeName.INT32, repetition);
+                trinoType = TINYINT;
+            }
+            case "short" -> {
+                typeBuilder = Types.primitive(PrimitiveType.PrimitiveTypeName.INT32, repetition);
+                trinoType = SMALLINT;
+            }
+            case "integer" -> {
+                typeBuilder = Types.primitive(PrimitiveType.PrimitiveTypeName.INT32, repetition);
+                trinoType = INTEGER;
+            }
+            case "long" -> {
+                typeBuilder = Types.primitive(PrimitiveType.PrimitiveTypeName.INT64, repetition);
+                trinoType = BIGINT;
+            }
+            case "float" -> {
+                typeBuilder = Types.primitive(PrimitiveType.PrimitiveTypeName.FLOAT, repetition);
+                trinoType = REAL;
+            }
+            case "double" -> {
+                typeBuilder = Types.primitive(PrimitiveType.PrimitiveTypeName.DOUBLE, repetition);
+                trinoType = DOUBLE;
+            }
+            case "boolean" -> {
+                typeBuilder = Types.primitive(PrimitiveType.PrimitiveTypeName.BOOLEAN, repetition);
+                trinoType = BOOLEAN;
+            }
+            case "binary" -> {
+                typeBuilder = Types.primitive(PrimitiveType.PrimitiveTypeName.BINARY, repetition);
+                trinoType = VARBINARY;
+            }
+            case "date" -> {
+                typeBuilder = Types.primitive(PrimitiveType.PrimitiveTypeName.INT32, repetition).as(LogicalTypeAnnotation.dateType());
+                trinoType = DATE;
+            }
+            case "timestamp" -> {
+                // Spark/DeltaLake stores timestamps in UTC, but renders them in session time zone.
+                // For more info, see https://delta-users.slack.com/archives/GKTUWT03T/p1585760533005400
+                // and https://cwiki.apache.org/confluence/display/Hive/Different+TIMESTAMP+types
+                typeBuilder = Types.primitive(PrimitiveType.PrimitiveTypeName.INT64, repetition).as(LogicalTypeAnnotation.timestampType(false, LogicalTypeAnnotation.TimeUnit.MILLIS));
+                trinoType = TIMESTAMP_MILLIS;
+            }
+            default -> throw new TypeNotFoundException(new TypeSignature(primitiveType));
```

Comment:
This should probably be `throw new TrinoException(NOT_SUPPORTED, format("Unsupported primitive type: %s", primitiveType))` because the type is not unknown to us, just that we don't support it yet.